ql/src/java/org/apache/hadoop/hive/ql/parse/ParseUtils.java

```diff
@@ -598,7 +598,26 @@ public static Map<Integer, List<ExprNodeGenericFuncDesc>> getFullPartitionSpecs(
       for (int i = 0; i < partSpecTree.getChildCount(); ++i) {
         CommonTree partSpecSingleKey = (CommonTree) partSpecTree.getChild(i);
         assert (partSpecSingleKey.getType() == HiveParser.TOK_PARTVAL);
-        String key = stripIdentifierQuotes(partSpecSingleKey.getChild(0).getText()).toLowerCase();
+        String columnName;
+        String key;
+        if (partSpecSingleKey.getChild(0).getType() == HiveParser.TOK_FUNCTION) {
+          int childCount = partSpecSingleKey.getChild(0).getChildCount();
+          if (childCount == 2) {  // Case with type and key
+            key = stripIdentifierQuotes(partSpecSingleKey.getChild(0).getChild(1).getText()).toLowerCase();
+            columnName = partSpecSingleKey.getChild(0).getChild(0).getText().toLowerCase() + "(" + key + ")";
+          } else if (childCount == 3) {  // Case with transform, columnName, and integer
+            key = stripIdentifierQuotes(partSpecSingleKey.getChild(0).getChild(2).getText()).toLowerCase();
+            String integerValue = partSpecSingleKey.getChild(0).getChild(1).getText();
```

Comment:
numBuckets?

Reply:
it ain't just in case of buckets, but also in case of truncate

Reply:
I will change it to ``transformParam`` as it is in ``TransformSpec``